docs/api.md

```diff
@@ -0,0 +1,551 @@
+mydoc = vr.body
+if vr.is_success:
+    download_vr: VantiqResponse = await client.download(mydoc.content)
+    if download_vr.is_success:
+        reader = download_vr.body
+        while True:
+            data = await reader.read(100)  # Read our data 100 bytes at a time
+            if len(data) == 0:
+                break
+                # Do something with the data
+                ...
+```
+
+### Vantiq.upload() (async) 
+Upload a file (or in-memory data), creating a object to hold that data.
+
+This allows the upload of a file to create a document, inage, video, or tensorflow model.
+
+#### Parameters
+   
+* resource : str -- The Vantiq resource type to create.
+* content_type : str -- The type of data contained in the file.  This will be set as the `contentType` of the resulting object.
+* filename : str -- The name of the file to upload.
+* doc_name : str -- (optional) The name of the object to create. If this is missing, use the filename.
+* inmem : str | bytes | bytearray -- (optional) Content to be uploaded.  Used with the content is not in a file.
+
+#### Returns
+`VantiqResponse`
+
+### Examples
+```python
+vr: VantiqResponse = await client.upload(VantiqResources.DOCUMENTS, 'iamge/png', '/file/name.png')
+if vr.is_success:
+    # Here, we will have uploaded a file named `/file/name.png`,
+    # and created a Document named `/file/name.png`.
+
+vr: VantiqResponse = await client.upload(VantiqResource.DOCUMENTS, 'text/plain', None,
+                                         'my document', 'some content for my document')
+if vr.is_success:
+    # Here, we will have created a document named 'my document' with the content
+    # 'some content for my document'
+
+```
+
+### Vantiq.count() (async)
+Return the number of items in a Vantiq resource that satisfy the where clause
+
+#### Parameters:
+
+* resource : str -- The name of the resource to be counted.  System resource names are provided via the VantiqResource class.
+* where : dict(str: \*) -- (optional) The "where clause" to be used to restrict the counting.  The where clause is defined in the [API Reference Guide](https://dev.vantiq.com/docs/system/api/index.html).
+
+#### Returns
+
+`VantiqResponse` --  where the `count` field contains the count requested.
+
+#### Vantiq.query() (async) 
+
+Send a query message to a Vantiq source
+
+#### Parameters
+
+* source_id : str --  Name of the source to which to send the query
+* query : dict -- The message describing the query to be sent.  These messages are source specific.
+
+#### Returns
+
+VantiqResponse indicating the success or failure of the query.  The body field will contain the results of the query (if applicable).
+
+
+### Vantiq.execute() (async)
+
+Execute a Vantiq procedure.
+
+#### Parameters:
+
+* procedure_id : str -- The name of the procedure to execute.
+* params : dict -- The parameters provided for the procedure's execution.  The key names are the parameter names, and the values their values.
```

Comment:
Do we plan to support array based execution (aka positional parameters)?

Reply:
Updated doc.  We do support that, it just didn't get incorporated here.